enclave-tls/samples/sgx-stub-enclave/sgx_stub_ecall.c

```diff
@@ -0,0 +1,284 @@
+		return -1;
+	}
+
+	int reuse = 1;
+	int ocall_ret = 0;
+	sgx_status = ocall_setsockopt(&ocall_ret,
+                                      sockfd,
+                                      ETLS_SOL_SOCKET,
+                                      ETLS_SO_REUSEADDR,
+                                      (const void *)&reuse,
+                                      sizeof(int));
+	if (sgx_status != SGX_SUCCESS || ocall_ret < 0) {
+		ETLS_ERR("Failed to call setsockopt() %#x %d\n",
+	                 sgx_status, ocall_ret);
+		return -1;
+	}
+
+	struct etls_sockaddr_in s_addr;
+	memset(&s_addr, 0, sizeof(s_addr));
+	s_addr.sin_family = ETLS_AF_INET;
+	s_addr.sin_addr.s_addr = s_ip;
+	s_addr.sin_port = s_port;
+
+	/* Bind the server socket */
+	sgx_status = ocall_bind(&ocall_ret,
+                                sockfd,
+                                (struct etls_sockaddr *)&s_addr,
+                                sizeof(s_addr));
+	if (sgx_status != SGX_SUCCESS || ocall_ret == -1) {
+		ETLS_ERR("Failed to call bind(), %#x %d\n",
+                         sgx_status, ocall_ret);
+		return -1;
+	}
+
+	/* Listen for a new connection, allow 5 pending connections */
+	sgx_status = ocall_listen(&ocall_ret, sockfd, 5);
+	if (sgx_status != SGX_SUCCESS || ocall_ret == -1) {
+		ETLS_ERR("Failed to call listen(), %#x %d\n",
+                       sgx_status, ocall_ret);
+		return -1;
+	}
+
+	/* Enclave-tls init */
+	libenclave_tls_init();
+	enclave_tls_handle handle;
+	enclave_tls_err_t ret = enclave_tls_init(&conf, &handle);
+	if (ret != ENCLAVE_TLS_ERR_NONE) {
+		ETLS_ERR("Failed to initialize enclave tls %#x\n", ret);
+		return -1;
+	}
+
+	/* Accept client connections */
+	struct etls_sockaddr_in c_addr;
+	uint32_t addrlen_in = sizeof(c_addr);
+	uint32_t addrlen_out;
+	while (1) {
+		ETLS_INFO("Waiting for a connection ...\n");
+
+		int64_t connd;
+		sgx_status = ocall_accept(&connd,
+                                          sockfd,
+                                          (struct sockaddr *)&c_addr,
+                                          addrlen_in,
+                                          &addrlen_out);
+		if (sgx_status != SGX_SUCCESS || connd < 0) {
+			ETLS_ERR("Failed to call accept() %#x %d\n",
+                                 sgx_status, connd);
+			return -1;
+		}
+
+		ret = enclave_tls_negotiate(handle, connd);
+		if (ret != ENCLAVE_TLS_ERR_NONE) {
+			ETLS_ERR("Failed to negotiate %#x\n", ret);
+			goto err;
+		}
+
+		ETLS_DEBUG("Client connected successfully\n");
+
+		char buf[256];
+		size_t len = sizeof(buf);
+		ret = enclave_tls_receive(handle, buf, &len);
+		if (ret != ENCLAVE_TLS_ERR_NONE) {
+			ETLS_ERR("Failed to receive %#x\n", ret);
+			goto err;
+		}
+
+		if (len >= sizeof(buf))
+			len = sizeof(buf) - 1;
+		buf[len] = '\0';
+
+		ETLS_INFO("Client: %s\n", buf);
+
+
+		/* Reply back to the client */
+		ret = enclave_tls_transmit(handle, buf, &len);
+		if (ret != ENCLAVE_TLS_ERR_NONE) {
```

Comment:
Server running in SGX Enclave will send mrenclave, mrsigner and hello message to client.
Is it necessary for server to send `mrenclave` and `mrsigner` to client in sgx enclave?

Reply:
it's a sample to send mrenclave and mrsigner